src/omv/imlib/framebuffer.c

```diff
@@ -170,23 +170,20 @@ void framebuffer_update_jpeg_buffer() {
                 bool overflow = false;
                 if (0) {
                 #if OMV_RAW_PREVIEW_ENABLE
-                } else if (src->is_mutable &&
-                           image_size(src) <= OMV_JPEG_BUFFER_SIZE_MAX) {
+                } else if ((src->pixfmt == PIXFORMAT_GRAYSCALE) || (src->pixfmt == PIXFORMAT_RGB565)) {
                     // Down-scale the frame (if necessary) and send the raw frame.
                     dst.size = src->bpp;
                     dst.pixfmt = src->pixfmt;
                     if (src->w <= OMV_RAW_PREVIEW_WIDTH && src->h <= OMV_RAW_PREVIEW_HEIGHT) {
                         memcpy(dst.pixels, src->pixels, image_size(src));
                     } else {
-                        dst.w = OMV_RAW_PREVIEW_WIDTH;
-                        dst.h = OMV_RAW_PREVIEW_HEIGHT;
-                        if (src->w > src->h) {
-                            dst.h = fast_floorf(dst.h * (src->h / (float) src->w));
-                        } else if (src->h > src->w) {
-                            dst.w = fast_floorf(dst.w * (src->w / (float) src->h));
-                        }
-                        imlib_draw_image(&dst, src, 0, 0, 1.0f, 1.0f, NULL, -1, 255,
-                                         NULL, NULL, IMAGE_HINT_BILINEAR, NULL, NULL, NULL);
+                        float x_scale = OMV_RAW_PREVIEW_WIDTH / (float) src->w;
+                        float y_scale = OMV_RAW_PREVIEW_HEIGHT / (float) src->h;
```

Comment:
```suggestion
                        float x_scale = OMV_RAW_PREVIEW_WIDTH / (float) src->w;
                        float y_scale = OMV_RAW_PREVIEW_HEIGHT / (float) src->h;
```
What if the source image is smaller than the preview image ? Let's say the preview size is 240x240 and the source is 120x120, wouldn't this up scale the frame to 240x240 ?

Reply:
No, the first if statement catches that. So, this only applies if the image is larger than the buffer.